cmd/utils.go

```diff
@@ -79,6 +86,194 @@ func parsePageRange(pageRange string) ([]int, error) {
 	return pages, nil
 }
 
+func parseInputPaths(inputPaths []string, recursive bool, matcher fileMatcher) ([]string, error) {
+	var err error
+	var files []string
+	acc := map[string]bool{}
+	lenInputPaths := len(inputPaths)
+
+	for i := 0; i < lenInputPaths; i++ {
+		// Convert relative paths to absolute ones.
+		inputPath := inputPaths[i]
+		if !filepath.IsAbs(inputPath) {
+			inputPath, err = filepath.Abs(inputPath)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Add visited file to the accumulator.
+		if _, ok := acc[inputPath]; ok {
+			continue
+		}
+		acc[inputPath] = true
+
+		// Get file info.
+		inputFile, err := os.Lstat(inputPath)
+		if err != nil {
+			return nil, err
+		}
+
+		// Check file type.
+		switch mode := inputFile.Mode(); {
+		case mode.IsRegular():
+			if matcher == nil || matcher(inputPath) {
+				files = append(files, inputPath)
+			}
+		case mode.IsDir():
+			dirFiles, err := parseInputDir(inputPath, recursive, acc, matcher)
+			if err != nil {
+				return nil, err
+			}
+			files = append(files, dirFiles...)
+		case mode&os.ModeSymlink != 0:
```

Comment:
In hindsight, it seems like the behavior of not resolving symlinks might be better.  Could be confusing if it starts going from /path/to/some/subdir and then starts exploring /usr/local or something recursively.   Initially I thought that the default behavior was to resolve symlinks automatically.  Whereas I think it would be fine if the symlinks are not resolved. My bad. Sorry about the confusion.